app/models/concerns/orchestration/dns.rb

```diff
@@ -42,7 +42,7 @@ def rebuild_dns
     results = {}
 
     DnsInterface::RECORD_TYPES.each do |record_type|
-      del_dns_record_safe(record_type)
+      dns_record(record_type).nil? || dns_record(record_type).valid? || del_dns_record_safe(record_type)
```

Comment:
The first check for `.nil?` is already done on `del_dns_record_safe` on `app/model/concerns/dns_interface.rb`.

Could you move the `.valid?` check to that method, and add a test on `test/models/orchestration/dns_test.rb` for rebuild_dns with an already correct DNS record?